### PR TITLE
Bugfix: don't call kubectl set resources when there would be no effect.

### PR DESCRIPTION
**Pull request by x13n**
Step towards fixing https://github.com/kubernetes/kubernetes/issues/61190, which is k8s 1.10.0 release blocker.

### CODE REVIEW

File: fluentd-gcp-scaler/scaler.sh [hunk 1]
```diff
@@ -80,6 +80,39 @@ build_flags() {
...
+  if needs_update requests.cpu ${CPU_REQUEST}; then NEED_UPDATE=true; fi
+  if needs_update requests.memory ${MEMORY_REQUEST}; then NEED_UPDATE=true; fi
+  if needs_update limits.cpu ${CPU_LIMIT}; then NEED_UPDATE=true; fi
+  if needs_update limits.memory ${MEMORY_LIMIT}; then NEED_UPDATE=true; fi
+  if ! ${NEED_UPDATE}
+  then
+    log "Nothing to update."
+    return
+  fi
+  if [ ${REQUESTS_FLAG} ] || [ ${LIMITS_FLAG} ]
```
Comment (kawych) [inline fluentd-gcp-scaler/scaler.sh]: Is it possible that code reaches this point and $REQUESTS_FLAG and/or $LIMIT_FLAG are empty?

Comment (x13n) [reply fluentd-gcp-scaler/scaler.sh]: Yes, but only under quite bizarre circumstances - when all values are not set anywhere (neither in defaults, nor in scaling policy), but resources are configured in the daemonset. The code above will figure it should update something, but both flags will be empty. Shouldn't happen in practice.

**Comment by x13n** [comment]
If there will be no further comments, I will merge it in 1h.

**Pull request merged**